testsuite/standalone/src/test/java/org/jboss/as/test/integration/security/ssl/sni/SNICombinedWithALPNTestCase.java

```diff
@@ -0,0 +1,403 @@
+import org.jboss.as.test.integration.management.util.ServerReload;
+import org.jboss.as.test.shared.PermissionUtils;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceActivator;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.ManagementClient;
+import org.wildfly.core.testrunner.ServerSetup;
+import org.wildfly.core.testrunner.ServerSetupTask;
+import org.wildfly.core.testrunner.WildflyTestRunner;
+import org.wildfly.security.x500.cert.X509CertificateBuilder;
+import org.wildfly.test.undertow.UndertowSSLService;
+import org.wildfly.test.undertow.UndertowSSLServiceActivator;
+import org.wildfly.test.undertow.UndertowServiceActivator;
+import org.xnio.OptionMap;
+import org.xnio.Xnio;
+import org.xnio.XnioWorker;
+import org.xnio.ssl.XnioSsl;
+
+import io.undertow.UndertowOptions;
+import io.undertow.client.ClientCallback;
+import io.undertow.client.ClientConnection;
+import io.undertow.client.ClientExchange;
+import io.undertow.client.ClientRequest;
+import io.undertow.client.UndertowClient;
+import io.undertow.protocols.ssl.UndertowXnioSsl;
+import io.undertow.server.DefaultByteBufferPool;
+import io.undertow.util.Headers;
+import io.undertow.util.Methods;
+import io.undertow.util.Protocols;
+import io.undertow.util.StringReadChannelListener;
+
+@RunWith(WildflyTestRunner.class)
+@ServerSetup(SNICombinedWithALPNTestCase.Setup.class)
+public class SNICombinedWithALPNTestCase {
+
+    public static final String SHA_256_WITH_RSA = "SHA256withRSA";
+    private static final String ALIAS = "server";
+    private static final String PASSWORD = "password";
+    private static final String[] HOST_KEY_STORE = {"subsystem", "elytron", "key-store", "host"};
+    private static final String[] IP_KEY_STORE = {"subsystem", "elytron", "key-store", "ip"};
+    private static final String[] HOST_KEY_MANAGER = {"subsystem", "elytron", "key-manager", "host"};
+    private static final String[] IP_KEY_MANAGER = {"subsystem", "elytron", "key-manager", "ip"};
+    private static final String[] HOST_SSL_CONTEXT = {"subsystem", "elytron", "server-ssl-context", "host"};
+    private static final String[] IP_SSL_CONTEXT = {"subsystem", "elytron", "server-ssl-context", "ip"};
+    private static final String[] SNI_SSL_CONTEXT = {"subsystem", "elytron", "server-ssl-sni-context", "test-context"};
+    private static final String TEST_JAR = "test.jar";
+
+    private static File hostNameKeystore;
+    private static File ipKeystore;
+
+    static class Setup implements ServerSetupTask {
+
+        @Override
+        public void setup(ManagementClient managementClient) throws Exception {
```

Comment:
Unless I am missing it, I don't see it in the setup the configuration of a 'token-realm'  with j'wt' stuff. Shouldn't that be covered by a test case (this one or another)?

Reply:
It may be appropriate to have a test added to WildFly for JWT, we don't have sufficient deployments within WildFly Core.